### ISSUE TITLE
Pixelated HLS stream

### ISSUE THREAD

**Issue by perlethandersen** [arut/nginx-rtmp-module#1145]
I have a server that streams HLS. If I stream from OBS to test it, it works fine, but when I stream from the needed source I get a stream that looks like this:

![wtfstream](https://user-images.githubusercontent.com/690450/32554318-b18d6fb4-c499-11e7-8ee7-d71bec314e69.PNG)

If I ffplay the stream directly via rtmp it looks like the video is h264.
Does anyone know what's wrong and how to fix it?

**Comment by perlethandersen** [comment]
Sorry for my, apparently, very basic question.
I didn't execute ffmpeg to transcode the stream for HLS. The reason I didn't do this at first was because it worked when I published from OBS. 

But yeah, it works now. However, I have noticed a large delay was added after using ffmpeg. My delay is right now around 25-27 seconds and was 7-10 seconds before.

What would be the fastest and best way to transcode the stream using ffmpeg?
